app/components/datepicker/DatepickerDirective.js

```diff
@@ -7,185 +7,184 @@
+                    showWeeks: false
+                };
+
+                vm.dateString = getFormattedDateString(vm.dateOptions.minDate,
+                                                        vm.dateOptions.maxDate);
+
+                vm.startDate = {
+                    opened: false
+                };
+
+                vm.endDate = {
+                    opened: false
+                };
+
+                vm.onChangeDatepicker = onChangeDatepicker;
+
+                vm.showDatepickerInfo = showDatepickerInfo;
+
+                vm.openEndDate = openEndDate;
+
+                vm.openStartDate = openStartDate;
+
+                vm.onSubmitDateText = onSubmitDateText;
+
+                vm.slider = defaultSliderValue();
+
+                /**
+                 * Set initial values for min and max dates in both of datepicker.
+                 */
+                vm.setInitialDates = function(){
+                    vm.dts = vm.dateOptions.minDate;
+                    vm.dte = vm.dateOptions.maxDate;
+                };
+
+                vm.setInitialDates();
+
+                scope.$on('setHistogram', setHistogram);
+
+                scope.$on('slideEnded', slideEnded);
+
+                /**
+                 * Will be called on click on start datepicker.
+                 * `minDate` will be reset to the initial value (e.g. 2000-01-01),
+                 * `maxDate` will be adjusted with the `scope.dte` value to
+                 *  restrict it not to be below the `minDate`.
+                 */
+                function openStartDate() {
+                    vm.startDate.opened = true;
+                    vm.dateOptions.minDate = vm.initialDateOptions.minDate;
+                    vm.dateOptions.maxDate = vm.dte;
+                }
+
+
+                /**
+                 * Will be called on click on end datepicker.
+                 * `maxDate` will be reset to the initial value (e.g. 2016-12-31),
+                 * `minDate` will be adjusted with the `scope.dts` value to
+                 *  restrict it not to be bigger than the `maxDate`.
+                 */
+                function openEndDate() {
+                    vm.endDate.opened = true;
+                    vm.dateOptions.maxDate = vm.initialDateOptions.maxDate;
+                    vm.dateOptions.minDate = vm.dts;
+                }
+
+                /**
+                 * Will be fired after the start and the end date was chosen.
+                 */
+                function onChangeDatepicker(){
+                    vm.dateString = getFormattedDateString(vm.dts, vm.dte);
+                    performDateSearch();
+                }
+
+                function getFormattedDateString(minDate, maxDate) {
+                    return '[' + minDate.toISOString().replace('.000Z','') + ' TO ' +
+                      maxDate.toISOString().replace('.000Z','') + ']';
+                }
+
+                function stringToDate(dateString) {
```

Comment:
can you rename it to `stringToStartEndDateArray` because it does return an Array with start and end date.